src/kili/mutations/asset/__init__.py

```diff
@@ -124,6 +126,7 @@ def append_many_to_dataset(
 
         if content_array is None and json_content_array is None:
```

Comment:
This condition can't happen since there is the decorator? there can be similar conditions in the code.

Reply:
the decorator only skips when content_array  and json_content_array are empty lists